### PR TITLE
{geo/data/devel}[intel/2015a] WRF 3.6.1, netCDF 4.3.2, netCDF-Fortran 4.4.0, CMake 3.0.0 + deps (WIP)

### PR DESCRIPTION
**Pull request by boegel**

### CODE REVIEW

File: easybuild/easyconfigs/c/CMake/CMake-3.0.0-intel-2015a.eb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+easyblock = 'ConfigureMake'
+
+name = 'CMake'
+version = '3.0.0'
```
Comment (wpoely86) [inline easybuild/easyconfigs/c/CMake/CMake-3.0.0-intel-2015a.eb]: 3.1.1?

File: easybuild/easyconfigs/h/HDF5/HDF5-1.8.13-intel-2015a.eb [hunk 2]
```diff
@@ -0,0 +1,26 @@
+name = 'HDF5'
+version = '1.8.13'
```
Comment (wpoely86) [inline easybuild/easyconfigs/h/HDF5/HDF5-1.8.13-intel-2015a.eb]: 1.8.14 is already merged?

File: easybuild/easyconfigs/j/JasPer/JasPer-1.900.1-intel-2015a.eb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+easyblock = 'ConfigureMake'
+
+name = 'JasPer'
+version = '1.900.1'
+
+homepage = 'http://www.ece.uvic.ca/~frodo/jasper/'
+description = """The JasPer Project is an open-source initiative to provide a free
+ software-based reference implementation of the codec specified in the JPEG-2000 Part-1 standard."""
+toolchain = {'name': 'intel', 'version': '2015a'} # tweaked by EasyBuild (was: {'name': 'intel', 'version': '2014b'})
```
Comment (wpoely86) [inline easybuild/easyconfigs/j/JasPer/JasPer-1.900.1-intel-2015a.eb]: newline above

File: easybuild/easyconfigs/n/netCDF-Fortran/netCDF-Fortran-4.4.0-intel-2015a.eb [hunk 4]
```diff
@@ -0,0 +1,18 @@
+name = 'netCDF-Fortran'
+version = '4.4.0'
+
+homepage = 'http://www.unidata.ucar.edu/software/netcdf/'
+description = """NetCDF (network Common Data Form) is a set of software libraries 
+ and machine-independent data formats that support the creation, access, and sharing of array-oriented 
+ scientific data."""
+
```
Comment (wpoely86) [inline easybuild/easyconfigs/n/netCDF-Fortran/netCDF-Fortran-4.4.0-intel-2015a.eb]: delete newline?

File: easybuild/easyconfigs/j/JasPer/JasPer-1.900.1-intel-2015a.eb (hunk 3 above)
Comment (boegel) [inline easybuild/easyconfigs/j/JasPer/JasPer-1.900.1-intel-2015a.eb]: ah crap, how did that get in here... :(

**Comment by wpoely86** [comment]
looks fine

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 13 out of 13 (9 easyconfigs in this PR)
Linux SL 6.5, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/e60e1a899b98de939380 for a full test report.

File: easybuild/easyconfigs/c/CMake/CMake-3.0.0-intel-2015a.eb (hunk 1 above)
Comment (boegel) [inline easybuild/easyconfigs/c/CMake/CMake-3.0.0-intel-2015a.eb]: this PR only includes existing easyconfigs with a new toolchain, so not touching versions of (build)dependencies

File: easybuild/easyconfigs/h/HDF5/HDF5-1.8.13-intel-2015a.eb (hunk 2 above)
Comment (boegel) [inline easybuild/easyconfigs/h/HDF5/HDF5-1.8.13-intel-2015a.eb]: this PR only includes existing easyconfigs with a new toolchain, so not touching versions of dependencies

File: easybuild/easyconfigs/n/netCDF-Fortran/netCDF-Fortran-4.4.0-intel-2015a.eb (hunk 4 above)
Comment (boegel) [inline easybuild/easyconfigs/n/netCDF-Fortran/netCDF-Fortran-4.4.0-intel-2015a.eb]: done

**Comment by gjbex** [comment]
Test report by @gjbex
**FAILED**
Build succeeded for 4 out of 9 (9 easyconfigs in this PR)
Linux red hat enterprise linux computenode 6.4, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.7.6
See https://gist.github.com/89d1be2533dd7842305c for a full test report.

**Comment by boegel** [comment]
@gjbex: not the result I was hoping for, but a known issue (sort of):

```
Couldn't find file HDF5-1.8.13_configure_intel.patch anywhere, and downloading it didn't work either...
```

This PR features several patches, so copying them would be painful.

I wonder why I didn't run into this though, hmm...

**Comment by boegel** [comment]
Probably explanation why @gjbex ran into the missing patch file in https://github.com/hpcugent/easybuild-framework/issues/1140#issuecomment-71277094 .

@gjbex is currently running a new `--from-pr --upload-test-report`, with the `easybuild-easyconfigs` location included in the robot search path (via `robot-paths` in cfg file) to confirm that this indeed resolves the issue.

**Comment by gjbex** [comment]
Test report by @gjbex
**FAILED**
Build succeeded for 11 out of 12 (9 easyconfigs in this PR)
Linux red hat enterprise linux computenode 6.4, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.7.6
See https://gist.github.com/b2cca626718ac7c37464 for a full test report.

**Comment by boegel** [comment]
WRF build at @gjbex's end failed with:

```
ERROR EasyBuild crashed with an error
(at easybuild/easybuild-framework/easybuild/tools/toolchain/mpi.py:205 in mpi_cmd_for):
Failed to create file /tmp/mpdboot: [Errno 1] Operation not permitted: '/tmp/mpdboot'
```

the `mpi_cmd_for` function is literally hardcoding `/tmp/mpdboot`, which is obviously a stupid idea, it should be using `tempfile.mkdtemp()`, at the very least.

I'll look into fixing this. Thanks for the test report @gjbex!

**Comment by boegel** [comment]
@wpoely86: can give you this a try on your end? The issue @gjbex ran into is orthogonal to this PR imho, but I'd like to see confirmation from someone else before we allow to merge this one in.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 9 out of 11 (9 easyconfigs in this PR)
Linux debian 7.8, Intel(R) Xeon(R) CPU            5130  @ 2.00GHz, Python 2.7.3
See https://gist.github.com/ee94ad89a0b6d45b1a32 for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 11 out of 11 (9 easyconfigs in this PR)
Linux debian 7.8, Intel(R) Xeon(R) CPU            5130  @ 2.00GHz, Python 2.7.3
See https://gist.github.com/51027a4ecbdd713b469b for a full test report.

**Comment by boegel** [comment]
@wpoely86: what happened with the failed JasPer build (see below)? Anything we can fix/avoid?

```
 icc -DHAVE_CONFIG_H -I. -I. -I../../../src/libjasper/include/jasper -I../../../src/libjasper/include -I/home/opt/easybuild/software/imkl/11.2.1.133-iimpi-7.2.3-GCC-4.9.2/mkl/include -fPIC -O2 -xHost -ftz -fp-speculation=safe -fp-model source -MT jpg_enc.lo -MD -MP -MF .deps/jpg_enc.Tpo -c jpg_enc.c -o jpg_enc.o
In file included from jpg_jpeglib.h(74),
                 from jpg_dec.c(75):
/usr/include/jpeglib.h(25): catastrophic error: cannot open source file "jconfig.h"
  #include "jconfig.h"		/* widely used configuration options */
```

Why is it using `/usr/include/jpeglib.h`?

**Comment by wpoely86** [comment]
@boegel the issue was that it clashes with a system package: https://packages.debian.org/en/wheezy/libjpeg8-dev

**Comment by boegel** [comment]
@wpoely86: do we have a separate ticket for following that up? is there any way in which this problem can be fixed?

**Comment by wpoely86** [comment]
Not that I know of.
I think it can be fixed. It's just a matter of priorities of the include directories, I *think*. Can be that the configure script pick up something it shouldn't.

I haven't looked at it that closely.

**Comment by boegel** [comment]
@wpoely86: ok, opened an issue to discuss/figure out a fix: #1331

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 5 out of 9 (1 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/e2983c59375376b66140 for a full test report.

**Comment by boegel** [comment]
ignore the failed test report, the test was run on our login nodes which limits forking of processes significantly, breaking the HDF5 build

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 9 out of 9 (9 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/020a18fdbce985e998ba for a full test report.

**Comment by boegel** [comment]
Retested on top of https://github.com/hpcugent/easybuild-framework/pull/1146, looks great now, so going in.

Thanks for testing/reviewing @wpoely86 and @gjbex!

**Pull request merged**